### PR TITLE
Change: switch game playing in the menu based on selected landscape

### PR DESCRIPTION
**Pull request by WimLeflere**
## Motivation

You only get an idea what the different landscapes look like once you load a game.
Showing the selected landscape type in the background of the menu gives a good idea what the differences are.

## Description

When a specific landscape is selected in the menu, a save file with that landscape type is loaded in the background of the menu.

## Limitations

Some lag (2-3 seconds) in debug build when switching landscape/save game.
Not really noticeable in a release build.

Save game's for arctic/tropical/toyland are not as interesting (yet) as the temperate/original save game.

### CODE REVIEW

File: src/openttd.cpp [hunk 1]
```diff
@@ -330,6 +330,24 @@ static void ShutdownGame()
 	UninitFreeType();
 }
 
+/**
```
Comment (perezdidac) [inline src/openttd.cpp]: Simplify in a single line:

/** Get the introduction game landscape specific save file */

File: src/openttd.cpp [hunk 2]
```diff
@@ -330,6 +330,24 @@ static void ShutdownGame()
...
+* Get the introduction game landscape specific save file.
+*/
+static std::string GetIntroGameSaveFile(LandscapeID landscape)
+{
+	switch (landscape)
+	{
+	case LT_ARCTIC:
```
Comment (perezdidac) [inline src/openttd.cpp]: As per the convention, please intend case blocks, and keep { in the same line
`switch (landscape) {`

**Comment by nielsmh** [comment]
On one hand I like the idea, on the other hand it means we would need to collect more saves (or run a rotation) when releasing new versions.

File: src/openttd.cpp (hunk 2 above)
Comment (WimLeflere) [reply src/openttd.cpp]: Has there been thought about adding a format file (.clang-format for example) to the repo?

**Comment by WimLeflere** [comment]
> On one hand I like the idea, on the other hand it means we would need to collect more saves (or run a rotation) when releasing new versions.

I first attempted to load the original file with a different landscape type to not have multiple save files.
But that doesn't seem to be supported/possible, or I didn't find the correct place to change is.

From going through the loading code I have the impression that old save game files are still supported.
So I don't think they will have to be updated (regularly).

**Comment by James103** [comment]
> On one hand I like the idea, on the other hand it means we would need to collect more saves (or run a rotation) when releasing new versions.

After this PR is merged, I propose the title screen competitions for OpenTTD 1.12 and beyond should allow for multiple entries per participant, but only one entry **per climate** per participant, instead of one entry per participant total. The entry and/or voting periods could be extended to accommodate this, and the entry with the most votes, **per climate**, wins.

File: src/openttd.cpp (hunk 2 above)
Comment (LordAro) [reply src/openttd.cpp]: I've personally tried it a few times. OTTD's codestyle is just too "unique" to be applied sanely, there's too much manually formatted code (the window definitions are the obvious examples)

**Comment by LC-Zorg** [comment]
Very interesting PR. Finally these buttons on the main menu would have some function. :) More saves are not a problem. I think even more participants will come, knowing they have a better chance. However, changing the background when creating a new game could no longer be appropriate.

**Comment by WimLeflere** [comment]
This is what the different menu landscapes look like.
arctic menu
toyland menu
tropical menu

**Comment by TrueBrain** [comment]
I love the idea and concept, but sadly, I do not consider it realistic for us to collect 4 savegames each release. And I really prefer a new title game for each release over this.

So I am really sorry, but I do not think we should include this. And not because it is a bad idea, but simply because we don't have the resources to have cool new games every release :(

**Pull request closed without merging**